cms/static/js/spec/views/modals/move_xblock_modal_spec.js

```diff
@@ -0,0 +1,73 @@
+        'use strict';
+        describe('MoveXBlockModal', function() {
+            var modal,
+                showModal,
+                DISPLAY_NAME = 'HTML 101',
+                OUTLINE_URL = '/course/cid?format=concise',
+                ANCESTORS_URL = '/xblock/USAGE_ID?fields=ancestorInfo';
+
+            showModal = function() {
+                modal = new MoveXBlockModal({
+                    sourceXBlockInfo: new XBlockInfo({
+                        id: 'USAGE_ID',
+                        display_name: DISPLAY_NAME,
+                        category: 'html'
+                    }),
+                    XBlockUrlRoot: '/xblock',
+                    outlineURL: OUTLINE_URL,
+                    XBlockAncestorInfoUrl: ANCESTORS_URL
+
+                });
+                modal.show();
+            };
+
+            beforeEach(function() {
+                setFixtures('<div id="page-notification"></div><div id="reader-feedback"></div>');
+                TemplateHelpers.installTemplates([
+                    'basic-modal',
+                    'modal-button',
+                    'move-xblock-modal'
+                ]);
+            });
+
+            afterEach(function() {
+                modal.hide();
+            });
+
+            it('rendered as expected', function() {
+                showModal();
+                expect(
+                    modal.$el.find('.modal-header .title').contents().get(0).nodeValue.trim()
+                ).toEqual('Move: ' + DISPLAY_NAME);
+                expect(
+                    modal.$el.find('.modal-sr-title').text().trim()
+                ).toEqual('Choose a location to move your component to');
+                expect(modal.$el.find('.modal-actions .action-primary.action-move').text()).toEqual('Move');
+            });
+
+            it('sends request to fetch course outline', function() {
+                var requests = AjaxHelpers.requests(this),
+                    renderViewsSpy;
+                showModal();
+                renderViewsSpy = spyOn(modal, 'renderViews');
+                expect(requests.length).toEqual(2);
+                AjaxHelpers.expectRequest(requests, 'GET', OUTLINE_URL);
+                AjaxHelpers.respondWithJson(requests, {});
+                AjaxHelpers.expectRequest(requests, 'GET', ANCESTORS_URL);
+                AjaxHelpers.respondWithJson(requests, {});
+                expect(renderViewsSpy).toHaveBeenCalled();
+            });
+
+            it('shows error notification when fetch course outline request fails', function() {
+                var requests = AjaxHelpers.requests(this),
+                    notificationSpy = ViewHelpers.createNotificationSpy('Error');
+                showModal();
+                AjaxHelpers.respondWithError(requests);
+                ViewHelpers.verifyNotificationShowing(notificationSpy, "Studio's having trouble saving your work");
+            });
+        });
+    });
```

Comment:
I'd also like to see a couple of Bok Choy tests:
 - a simple end-to-end test that moves a block and then verifies that it moved to the right place
 - an a11y test that verifies that the modal is accessible